### DISCUSSION TITLE
RLS not working

### DISCUSSION THREAD

**Discussion by PMarco03** [supabase/supabase#21382] [Questions] [answered]
Hi, I created a Vue app with Supabase to store various oils with different attributes. I used Supabase's auth service to manage accounts. I want to grant access to specific users for specific tables using RLS. This was my idea: I have 22 tables in my database. I created a table called "tables_list" that contains all my database tables. Additionally, I have a table called "user_permissions" that links the records from "tables_list" with the default table of Supabase auth.users.

I created a function called "has_permissions(uid, text)" that takes as arguments the user UID and the table name, and checks if the specific user is associated with the specific table:

  ```
  RETURN EXISTS (
        SELECT 1
        FROM public.user_permissions
        JOIN auth.users ON auth.users.id = userid
        JOIN public.tables ON public.tables.id = public.user_permissions.cod_tables
        WHERE public.tables.name = tablename AND auth.users.id = userid
    );
```

So, if I want to apply this function to a RLS (Row-Level Security) policy for a table, for example, one called "attributes_list1", I should write this:

```
CREATE POLICY "permissions"
ON "public"."attributes_list1"
TO authenticated
USING (has_permissions(auth.uid(), 'attributes_list1'::text));
```

However, when I try to fetch the tables, it returns an error because I don't have permissions (it works if I disable RLS). I tried calling the function with different arguments, and it works fine. What am I doing wrong?

**Comment by GaryAustin1** [comment]
Your function would need to be security definer type to access auth and bypass any other RLS on tables.

**Reply by PMarco03** [reply]
Hi! So, what should i do?

**Reply by GaryAustin1** [reply, marked as answer]
Make your function security definer type.  https://supabase.com/docs/guides/auth/row-level-security#using-security-definer-functions
